client/web/src/repo/settings/RepoSettingsOptionsPage.tsx

```diff
@@ -81,12 +113,47 @@ export const RepoSettingsOptionsPage: FC<Props> = ({ repo, history }) => {
                         ))}
                         {services.length > 1 && (
                             <>
-                                <Text className="text-muted">
-                                    This repository is mirrored from multiple code hosts. To remove the repository from
-                                    this Sourcegraph instance, all code host configurations need to be updated.
+                                <Text>
+                                    This repository is mirrored by multiple code hosts. To change access, disable, or
+                                    remove this repository, the configuration must be updated on all code hosts.
                                 </Text>
+                                <Button
+                                    variant="primary"
+                                    className={styles.button}
+                                    onClick={event => {
+                                        event.preventDefault()
+                                        setExclusionInProgress(true)
+                                        excludeRepo({
+                                            variables: {
+                                                externalServices: services.map(svc => svc.id),
+                                                repo: repo.id,
+                                            },
+                                        })
+                                            .catch(
+                                                // noop here is used because update error is handled directly when useMutation is called
+                                                noop
+                                            )
+                                            .finally(() => {
+                                                setExclusionInProgress(false)
+                                            })
+                                    }}
+                                    disabled={excludingDisabled || (exclusionInProgress && !isExcluding)}
+                                >
+                                    <span className={exclusionInProgress ? styles.invisibleText : ''}>
+                                        Exclude repository from all code hosts
+                                    </span>
+                                    {exclusionInProgress && <LoadingSpinner className={styles.loader} />}
```

Comment:
Not sure about the tradeoffs here, but thought that this could potentially be a ternary if statement:

```suggestion
                                    {exclusionInProgress
                                    ? <LoadingSpinner className={styles.loader} />
                                    : <span>Exclude repository from all code hosts</span>}
```

Reply:
The thing here is that I've added this weird <span> with `visibility: hidden` so that the size of the button doesn't change during loading.

It used to be a text-wide button, which collapses to spinner-wide button. One of the least hacky options I found is to do this. Button size is constant because it is always a text-wide button, we just don't show the text when we need to show a spinner 😁